validator/client/src/main/java/tech/pegasys/teku/validator/client/DefaultValidatorStatusLogger.java

```diff
@@ -19,48 +19,66 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
 import tech.pegasys.teku.api.response.v1.beacon.ValidatorStatus;
 import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
 import tech.pegasys.teku.validator.api.ValidatorApiChannel;
 
 public class DefaultValidatorStatusLogger implements ValidatorStatusLogger {
 
   private static final int VALIDATOR_KEYS_PRINT_LIMIT = 20;
+  private static final long INITIAL_STATUS_CHECK_RETRY_PERIOD = 5; // seconds
 
   final List<BLSPublicKey> validatorPublicKeys;
   final ValidatorApiChannel validatorApiChannel;
   final AtomicReference<Map<BLSPublicKey, ValidatorStatus>> latestValidatorStatuses =
       new AtomicReference<>();
+  final AsyncRunner asyncRunner;
 
   public DefaultValidatorStatusLogger(
-      List<BLSPublicKey> validatorPublicKeys, ValidatorApiChannel validatorApiChannel) {
+      List<BLSPublicKey> validatorPublicKeys,
+      ValidatorApiChannel validatorApiChannel,
+      AsyncRunner asyncRunner) {
     checkArgument(!validatorPublicKeys.isEmpty());
     this.validatorPublicKeys = validatorPublicKeys;
     this.validatorApiChannel = validatorApiChannel;
+    this.asyncRunner = asyncRunner;
   }
 
   @Override
-  public void printInitialValidatorStatuses() {
-    validatorApiChannel
+  public SafeFuture<Void> printInitialValidatorStatuses() {
+    return validatorApiChannel
         .getValidatorStatuses(validatorPublicKeys)
-        .thenAccept(
+        .thenCompose(
```

Comment:
I recently added a retry utility (`AsyncRunner.runWithRetry`) that might be useful here.